packages/picasso-lab/src/TextEditor/hooks/useTextEditor.tsx

```
@@ -0,0 +1,93 @@
+import { useReducer, useCallback } from 'react'
+
+import { Props as TextEditorProps } from '../TextEditor'
+import { ToolbarStateType } from '../types'
+import useAutofocus from './useAutofocus'
+import useDisableEditor from './useDisableEditor'
+import useEditorLooseFocusFix from './useEditorLooseFocusFix'
+import useOnChange from './useOnChange'
+import useOnValueChange from './useOnValueChange'
+import useQuillInstance from './useQuillInstance'
+import useToolbarActiveStates from './useToolbarActiveStates'
+import useToolbarHandlers from './useToolbarHandlers'
+
+type ActionType = {
+  type: 'set'
+  payload: ToolbarStateType
+}
+
+type Props = {
+  id: TextEditorProps['id']
+  placeholder: TextEditorProps['placeholder']
+  onChange: TextEditorProps['onChange']
+  value: TextEditorProps['value']
+  autofocus: TextEditorProps['autofocus']
+  disabled: TextEditorProps['disabled']
+}
+
+const initialState: ToolbarStateType = {}
+
+const reducer = (
+  state: ToolbarStateType,
+  { type, payload }: ActionType
+): ToolbarStateType => {
+  switch (type) {
+    case 'set':
+      return payload
+
+    default:
+      console.log('unknown action type')
```

bad handling of the state change 😃 better to throw the error